### ISSUE TITLE
SmartOS: able to open same DB twice

### ISSUE THREAD

**Issue by slfritchie** [basho/eleveldb#62]
See the "slf-bbot-test-fixing1" branch.

The `tests/cleanup.erl` test `assumption_test()` is failing by allowing the same database instance to be opened twice.  This test passes (i.e., DB cannot be opened twice) on OS X and Linux.  It fails on SmartOS.

Example output, note that the `erlang:display()` output from line 138 does not appear.

```
module 'cleanup'
  cleanup: assumption_test...
{line,36}
{line,128}
{line,131}
{line,38}
{line,135}
{line,128}   %% This is output from the next EUnit test
*failed*
::{badmatch,{ok,<<>>}}
```

**Comment by slfritchie** [comment]
Links:
- top of `assumption_test()`: https://github.com/basho/eleveldb/blob/slf-bbot-test-fixing1/test/cleanup.erl#L35
- top of `failed_open()`: https://github.com/basho/eleveldb/blob/slf-bbot-test-fixing1/test/cleanup.erl#L134

**Comment by matthewvon** [comment]
mv-google-filelock ... this fix should have already been in the code base for 1.3.0.  Somehow I missed it.  Of course, not sure why old code is failing on SmartOS since it was hand tested at Voxer.

**Comment by matthewvon** [comment]
P.S. this is a leveldb fix, not eleveldb.

**Comment by slfritchie** [comment]
Fixed by commit 91fbe5c407, closing.
